### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by bovine3dom**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: "Code scanning - action"
+
+on:
+  push:
+    branches: 'master'
```
Comment (bovine3dom) [inline .github/workflows/codeql-analysis.yml:5]:
```suggestion
    branches: '*'
```

**Comment by bovine3dom** [comment]
I don't think I care about this enough to debug it :)

**Comment by glacambre** [comment]
@bovine3dom Are there specific queries you wanted to run or did you just want to add a generic static analyzer? The former can be done on lgtm.com and the latter can be done by adding the lgtm bot to Tridactyl.

**Comment by bovine3dom** [comment]
GitHub said there was a shiny new thing so I thought I would try it out, so I suppose it's closer to the latter :)

**Pull request closed without merging**